1-mandatory/3-slideshow/slideshow.js

```diff
@@ -1 +1,64 @@
-// Write your code here
+let slideshowImages = [
+  "./imgs/1.jpeg",
+  "./imgs/2.jpeg",
+  "./imgs/3.jpeg",
+  "./imgs/4.jpeg",
+  "./imgs/5.jpeg",
+];
+const autoBackButton = document.querySelector("#auto-back");
+const backButton = document.querySelector("#back");
+const stopButton = document.querySelector("#stop");
+const forwardButton = document.querySelector("#forward");
+const autoForwardButton = document.querySelector("#auto-forward");
+const mainImg = document.querySelector("#img");
+const commentImg = document.querySelector("#photoIndex");
+let photoNumber = 0;
+let intervalTime;
+let intervalIndex;
+
+function forwardAction() {
+  if (photoNumber === slideshowImages.length) {
+    photoNumber = 0;
+  }
+  photoNumber++;
+  commentImg.innerText = photoNumber;
+  mainImg.src = slideshowImages[photoNumber - 1];
+}
+function backAction() {
+  if (photoNumber === 0) {
+    photoNumber = slideshowImages.length + 1;
+  }
+  if (photoNumber === 1) {
+    photoNumber = slideshowImages.length + 1;
+  }
+  photoNumber--;
+  commentImg.innerText = photoNumber;
+  mainImg.src = slideshowImages[photoNumber - 1];
+}
```

Comment:
All good! I would suggest creating a function to update the image and index: Instead of duplicating the code to update the image and index in the forwardAction and backAction functions, you could create a separate function to handle this logic. This will make the code more modular and easier to maintain. This seperate function will handle updating the main image and the index in the comment section. Then, I would call this function inside the `forwardAction` and `backAction` functions, passing in the appropriate parameters.